nextcord/http.py

```diff
@@ -269,152 +574,185 @@ async def request(
+                                #  current RateLimit, finish up and signal that the current bucket should be migrated
+                                #  to the stored one.
+                                _log.debug(
+                                    "Route %s with bucket %s already exists, migrating other possible requests to "
+                                    "that bucket.",
+                                    rate_limit_path,
+                                    url_rate_limit.bucket,
                                 )
-                                self._global_over.clear()
-                            else:
-                                self._dispatch(
-                                    "http_ratelimit",
-                                    int(limit),
-                                    int(remaining),
-                                    retry_after,
-                                    bucket,
-                                    response.headers.get("X-RateLimit-Scope"),
+                                correct_rate_limit = self._buckets[url_rate_limit.bucket]
+                                self._set_url_rate_limit(
+                                    route.method, route.path, auth, correct_rate_limit
                                 )
+                                # Update the correct RateLimit object with our findings.
+                                await correct_rate_limit.update(response)
+                                if correct_rate_limit.bucket:
+                                    # Signals to all requests waiting to acquire to migrate.
+                                    url_rate_limit.migrate_to(correct_rate_limit.bucket)
+                                else:
+                                    raise ValueError(
+                                        "Migrating to bucket %s, but correct_rate_limit.bucket is falsey. This is "
+                                        "likely an internal Nextcord issue and should be reported."
+                                    )
+                            elif url_rate_limit.bucket is not None:
+                                self._buckets[url_rate_limit.bucket] = url_rate_limit
+
+                            # even errors have text involved in them so this is safe to call
+                            ret = await json_or_text(response)
+
+                            # self._dispatch(
+                            #     "http_ratelimit",  # TODO: Uhhhhhhhhhhh
+                            #     int(limit),
+                            #     int(remaining),
+                            #     delta,
+                            #     bucket,
+                            #     response.headers.get("X-RateLimit-Scope"),
+                            # )
+
+                            if response.status >= 400:
+                                if response.status in {500, 502, 504}:
```

Comment:
Couldnt this just check if its 500 or higher

Reply:
The `if response.status in {500, 502, 504}` was lifted from the original `HTTPClient.request()` code, changing it to `>= 500` is a good idea.

Reply:
Hmm thats true, fair enough